server/task_boltdb.go

```diff
@@ -221,32 +225,39 @@ func loadTaskLogs(tx *bolt.Tx, task *tes.Task) {
 // GetTask gets a task, which describes a running task
 func (taskBolt *TaskBolt) GetTask(ctx context.Context, req *tes.GetTaskRequest) (*tes.Task, error) {
 	var task *tes.Task
-	err := taskBolt.db.View(func(tx *bolt.Tx) error {
-		task = getTaskView(tx, req.Id, req.View)
-		return nil
+	var err error
+	err = taskBolt.db.View(func(tx *bolt.Tx) error {
+		task, err = getTaskView(tx, req.Id, req.View)
+		return err
 	})
-	return task, err
+	if err != nil {
```

Comment:
This will mark every error as a missing task, which will be misleading for an error at some point. You can get around this by returning an errTaskNotFound  value from loadBasicTaskView, instead of a generic fmt.Errorf.

Also, I may be reading the code wrong, but I think this will still incorrectly return a task message for a non-existant task when using the MINIMAL view.